### ISSUE TITLE
Problems getting numeric-comp citation style to work

### ISSUE THREAD

**Issue by claczny** [jgm/pandoc#2065]
Hi,

I have a set of MarkDown files in which I use references à la `[@REF1; @REF2; @REF3]`. When I run `pandoc` directly on the MarkDown files to produce a PDF output with the Nature CSL, I get them nicely formatted into `[REF1 - REF3]`.

In parallel, I am also creating a TeX output that I am running through `latexmk`. However, despite specifying

```
$if(biblatex)$
\usepackage[
    citestyle=numeric-comp,
    style=numeric,
    isbn=false,
    doi=false,
    sorting=none,
    autocite=inline
]{biblatex}
```

in a custom template file (derived from the original `default.latex`), the references are not compacted but are listed as `[REF1, REF2, REF3]`.

This seems to be related to `pandoc` creating `\autocites{REF1}{REF2}{REF3}` which apparently does not allow the references to be compacted. If I manually edit that to `\cite{REF1, REF2, REF3}` the result is the desired `[REF1 - REF3]` also when using `latexmk`.

It may appear somewhat counterintuitive to produce a PDF via `pandoc` **and in parallel** via `latexmk` but I am nevertheless wondering how to get the `latexmk` output to match the `pandoc` output.

Looking forward to your clarifications.

Best,

Cedric

**Comment by lierdakil** [comment]
A quick question, does `\autocites{REF1, REF2, REF3}` work?

**Comment by claczny** [comment]
@lierdakil Yes, that works, thank you!
I use the following sed command to replace the curly brackets `sed -i '/\\autocites/ s/}{/,/g' mytexfile.tex`.
Could you think of a way that is 'pandoc-only', i.e., no need to call `sed`?

**Comment by lierdakil** [comment]
I believe you actually stumbled on a bug (or you might say a deficiency), since, by all accounts, this should be the default behavior. It should be possible to borrow algorithms from pandoc-citeproc, so I'll make a point to take a look at this when I get some extra time on my hands. For now, using sed will have to do though.

**Comment by lierdakil** [comment]
So, I've taken a look, and here's what I came up with at the moment.

Citations can be grouped together by matching prefix and suffix, e.g.

```markdown
[@REF1;Stevens at @REF2 on p. 44; @REF3 on p. 42; Johns in @REF4; @REF5 on p. 42; @REF6]
```

will translate to

```latex
\autocites{REF1,REF6}[Stevens at][ on p.~44]{REF2}[ on
p.~42]{REF3,REF5}[Johns in][]{REF4}
```

and should be rendered as something like (with `citestyle=numeric-comp,style=numeric,autocite=inline`)

```
[1, 2, Stevens at 3, on p. 44, 4, 5, on p. 42, Johns in 6]
```

In some border cases this could be suboptimal though, esp. with suffixes. Maybe I should group only by prefixes, and, if suffixes are not null, then create separate groups? I.e. `[@REF1 p. 21; @REF2 p. 21]` into `[1, 2 p. 21]` vs `[1, p. 21, 2, p. 21]`.

**Comment by jgm** [comment]
This should have been closed by #2091.

**Comment by lierdakil** [comment]
Except #2091 was never merged due to me never finding the way to make it work (sorry about that)
